x/vesting/keeper/msg_server.go

```diff
@@ -87,10 +90,22 @@ func (k Keeper) CreateClawbackVestingAccount(
 		case msg.FromAddress != va.FunderAddress:
 			return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "account %s can only accept grants from account %s", msg.ToAddress, va.FunderAddress)
 		}
-		k.AddGrantToClawbackVestingAccount(ctx, va, msg.GetStartTime(), msg.GetLockupPeriods(), msg.GetVestingPeriods(), vestingCoins)
+		k.addGrant(ctx, va, msg.GetStartTime(), msg.GetLockupPeriods(), msg.GetVestingPeriods(), vestingCoins)
 	} else {
-		baseAccount := ak.NewAccountWithAddress(ctx, to)
-		va = types.NewClawbackVestingAccount(baseAccount.(*authtypes.BaseAccount), from, vestingCoins, msg.StartTime, msg.LockupPeriods, msg.VestingPeriods)
+		acc := ak.NewAccountWithAddress(ctx, to)
+		ethAccount, ok := acc.(*etherminttypes.EthAccount)
+		if !ok {
+			return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "invalid account type: %T", ethAccount)
+		}
```

Comment:
we can used the original code here, no need to create a new eth account

Reply:
@fedekunze Without the modification I always got an interface conversion error, because `ak.NewAccountWithAddress` returns an `EthAccount` and `NewClawbackVestingAccount expects a `BaseAccount`.

Reply:
I see, then I'd just create the `BaseAccount` manually